### PR TITLE
[receiver/prometheus] Enable pkg.translator.prometheus.NormalizeName by default

### LINKED ISSUE #20518: Enable pkg.translator.prometheus.NormalizeName by default
### Component(s)

receiver/prometheus

### Describe the issue you're reporting

This feature gate has been in alpha for some time, this issue is to track any work needed to address before moving it to beta

### PR DESCRIPTION
**Pull request by codeboten**
Fixes #20518

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by kovrus — APPROVED** [review]

File: .chloggen/codeboten_beta-prom-normalize.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: prometheusreceiver
```
Comment (kovrus) [inline .chloggen/codeboten_beta-prom-normalize.yaml:5]: That is also relevant to Prometheus exporters? (tests were adjusted exporter/prometheusexporter/collector_test.go)The Prometheus and Prometheus remote write exporters will be producing different names for exporter series now? Shall we provide some migration instructions for users like how to keep the old behavior, etc.?

**Review by gouthamve — APPROVED** [review]
LGTM

File: .chloggen/codeboten_beta-prom-normalize.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Enable pkg.translator.prometheus.NormalizeName by default
+
+# One or more tracking issues related to the change
+issues: [20518]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
Comment (codeboten) [inline .chloggen/codeboten_beta-prom-normalize.yaml:16]:
```suggestion
subtext: |
  Enabling normalization by default changes the emitted metrics for components that rely on
  the prometheus translation package. This feature can still be disabled via the following
  `--feature-gates=-pkg.translator.prometheus.NormalizeName`
```

File: .chloggen/codeboten_beta-prom-normalize.yaml (hunk 1 above)
Comment (codeboten) [inline .chloggen/codeboten_beta-prom-normalize.yaml:5]:
```suggestion
component: prometheusreceiver, prometheusexporter, prometheusremotewrite
```

Comment (codeboten) [reply .chloggen/codeboten_beta-prom-normalize.yaml]: I've updated the changelog to include more details about this.

**Review by kovrus — APPROVED** [review]

**Pull request merged**